flask_praetorian/base.py

```diff
@@ -721,25 +740,46 @@ def read_token_from_cookie(self):
         return self._unpack_cookie(flask.request.cookies)
 
     def read_token(self):
-        exc = None
-        if 'header' in self.jwt_places:
-            try:
-                return self.read_token_from_header()
-            except MissingToken as e:
-                exc = e
-        if 'cookie' in self.jwt_places:
+        """
+        Tries to unpack the token from the current flask request
+        in the locations configured by JWT_PLACES.
+        Check-Order is defined by the value order in JWT_PLACES.
+        """
+
+        for place in self.jwt_places:
             try:
-                return self.read_token_from_cookie()
-            except MissingToken as e:
-                exc = e
-        if exc:
-            raise MissingToken("JWT token not found in {}".format(
-                self.jwt_places))
+                return getattr(
+                    self,
+                    "read_token_from_{place}".format(
+                        place=place.lower(),
+                    ),
+                )()
+            except MissingToken:
+                pass
+            except AttributeError:
+                flask.current_app.logger.warning(
+                    "Flask_Praetorian hasn't implemented reading JWT tokens from location {place}. Please reconfigure JWT_PLACES.".format(  # noqa: E501
```

Comment:
This is a good place to use dedent:
```suggestion
                    snick.dedent(
                        f"""
                        Flask_Praetorian hasn't implemented reading JWT tokens from location {place.lower}.
                        Please reconfigure JWT_PLACES.
                        """
                    )
```

Reply:
used the alreay imported texwrap module for this